.clang-format

```diff
@@ -0,0 +1,155 @@
+---
```

Comment:
I would suggest adding this `.clang-format` configuration in a separate PR. It looks like it altered a handful of files that are unrelated to this PR's central focus.

Reply:
That makes sense I will go ahead and do that now